tree/ntuple/v7/inc/ROOT/RNTupleView.hxx

```diff
@@ -164,7 +164,13 @@ private:
    RNTupleView(DescriptorId_t fieldId, Detail::RPageSource* pageSource)
      : fField(pageSource->GetDescriptor().GetFieldDescriptor(fieldId).GetFieldName()), fValue(fField.GenerateValue())
    {
-      Detail::RFieldFuse::ConnectRecursively(fieldId, *pageSource, fField);
+      fField.SetOnDiskId(fieldId);
+      fField.ConnectPageStorage(*pageSource);
```

Comment:
It is not clear (and/or I missed the description) why `SetOnDiskId` is called along side `ConnectPageStorage
` and why sometimes only `ConnectPageStorage` is called.  Could you clarify?

Reply:
The separation of `SetOnDiskId` and `ConnectPageStorage` allows for lazy/late binding. We can create fields and remember to which field ID on disk they refer to but only acquire actual I/O buffers when we really need the field.  That's handy for RDF, for example.  When an RNTuple view is created, however, the underlying field should immediately be available for reading, so we call both `SetOnDiskId` and `ConnectPageStorage` at the same time.